badge_server/test_badges.py

```diff
@@ -786,5 +822,411 @@ def test_github_py2py3_py3_incompatible_fresh_nodeps(self):
+            'installed_version_time': datetime.datetime(
+                2019, 2, 19, 21, 15, 56),
+            'is_latest': False,
+            'latest_version': '1.6.3',
+            'latest_version_time': datetime.datetime(
+                2019, 2, 19, 21, 15, 56)
+        }
+
+        old_dep_compat_results = list(RECENT_SUCCESS_DATA)
+        old_dep_compat_results.remove(GOOGLE_API_CORE_RECENT_SUCCESS_2)
+        old_dep_compat_results.remove(GOOGLE_API_CORE_RECENT_SUCCESS_3)
+        old_dep_compat_results.append(
+            GOOGLE_API_CORE_RECENT_SUCCESS_2.with_updated_dependency_info(
+                old_dep_info))
+        old_dep_compat_results.append(
+            GOOGLE_API_CORE_RECENT_SUCCESS_3.with_updated_dependency_info(
+                old_dep_info))
+
+        self.fake_store.save_compatibility_statuses(old_dep_compat_results)
+        package_name = 'google-api-core'
+        json_response = self.get_image_json(package_name)
+        self.assertEqual(json_response['left_text'],
+                         'compatibility check (PyPI)')
+        self.assertEqual(json_response['right_text'], 'old dependency')
+        self.assertEqual(json_response['right_color'], '#A4A61D')
+        self.assertLinkUrl(package_name, json_response['whole_link'])
+
+    def test_git_py2py3_off_by_patch(self):
+        old_dep_info = dict(UP_TO_DATE_DEPS)
+        old_dep_info['google-auth'] = {
+            'current_time': datetime.datetime.utcnow(),
+            'installed_version': '1.6.0',
+            'installed_version_time': datetime.datetime(
+                2019, 2, 19, 21, 15, 56),
+            'is_latest': False,
+            'latest_version': '1.6.3',
+            'latest_version_time': datetime.datetime(
+                2019, 2, 19, 21, 15, 56)
+        }
+
+        old_dep_compat_results = list(RECENT_SUCCESS_DATA)
+        old_dep_compat_results.remove(GOOGLE_API_CORE_GIT_RECENT_SUCCESS_2)
+        old_dep_compat_results.remove(GOOGLE_API_CORE_GIT_RECENT_SUCCESS_3)
+        old_dep_compat_results.append(
+            GOOGLE_API_CORE_GIT_RECENT_SUCCESS_2.with_updated_dependency_info(
+                old_dep_info))
+        old_dep_compat_results.append(
+            GOOGLE_API_CORE_GIT_RECENT_SUCCESS_3.with_updated_dependency_info(
+                old_dep_info))
+
+        self.fake_store.save_compatibility_statuses(old_dep_compat_results)
+        package_name = 'git+git://github.com/google/api-core.git'
+        json_response = self.get_image_json(package_name)
+        self.assertEqual(json_response['left_text'],
+                         'compatibility check (master)')
+        self.assertEqual(json_response['right_text'], 'old dependency')
+        self.assertEqual(json_response['right_color'], '#A4A61D')
+        self.assertLinkUrl(package_name, json_response['whole_link'])
+
+
+class TestBadgeImageObsoleteDependency(BadgeTestCase):
+    """Tests for cases where the badge image displays 'obsolete dependency.'"""
```

Comment:
```suggestion
    """Tests for cases where the badge image displays 'obsolete dependency'."""
```

Super nit ;-)

Reply:
Oh it's muscle memory for me to to put the punctuation inside the quotations.